src/stitch.rs

```diff
@@ -54,10 +55,10 @@ impl IterStitchedIndexHunks {
     /// the same point in the previous band, continuing backwards recursively
     /// until either there are no more previous indexes, or a complete index
     /// is found.
-    pub(crate) fn new(archive: &Archive, band_id: &BandId) -> IterStitchedIndexHunks {
+    pub(crate) fn new(archive: &Archive, band_id: Option<BandId>) -> IterStitchedIndexHunks {
```

Comment:
Passing none will result in an iterator only returning `None`.  
This is intentional thought, since it's simplifying the logic for `BackupWriter`.  
There is no need to check if there had been a band beforehand.

Reply:
Sounds good, and please add it to the docstring.

Reply:
:+1: Give me a few min. Writing a test rn.